cmd/app.go

```diff
@@ -433,16 +433,25 @@ func appUpgrade(ctx *cli.Context) error {
 	}
 
 	answers := app.Answers
-	answers, err = processAnswers(ctx, c, nil, answers, false)
-	if err != nil {
-		return err
+	if ctx.String("answers") != "" {
+		answers, err = processAnswers(ctx, c, nil, answers, false)
+		if err != nil {
+			return err
+		}
+	}
+
+	values := app.ValuesYaml
+	if ctx.String("values") != "" {
+		values, err = parseFileToString(ctx.String("values"))
+		answers = nil
```

Comment:
It looks like the helm controller accepts both, answers and values and appends the passed answers to values as per https://github.com/rancher/rancher/blob/master/pkg/controllers/user/helm/common/common.go#L216-L240
So we don't need to set answers to nil here?

Reply:
based on how the ui works - we should return answers or values, not both